### PR TITLE
[core] File name in IO is wstring instead of string

### PR DESCRIPTION
**Pull request by maceligueta**
This is a first attempt to allow loading files with strange characters in their name. It does not allow to read contents in the file with these characters, only in the file name.
It was discussed for a long time in #1511.
I've manually tested file names with accents and symbols like º, for example.
As this is a very delicate spot in Kratos, I think we could work a bit on this PR and add some tests which prove that it works and nothing got broken. Suggestions are welcome, as always.

### CODE REVIEW

File: kratos/utilities/timer.cpp [hunk 1]
```diff
@@ -140,6 +140,20 @@ int Timer::SetOuputFile(std::string const& rOutputFileName)
     return msOutputFile.is_open();
 }
 
+int Timer::SetOuputFile(std::wstring const& rOutputFileName)
```
Comment (maceligueta) [inline kratos/utilities/timer.cpp:143]: Here I did not manage to template this function, though indeed I tried.

**Comment by philbucher** [comment]
Why again was this necessary?
I got a bit lost in the discussion in #1511

**Comment by maceligueta** [comment]
It is quite common that Kratos users decide to save their models with names which include non-ascii characters. At least we noticed this in Windows. It happened in the past accents or symbols like º. For example, you want to save your model saying that the orientation of the wall was 30º.
With the current version of Kratos, it crashes. It just says that it can't open the mdpa file. This should fix it.

**Comment by maceligueta** [comment]
It is true that the discussion in #1511 got messed up a bit because it was considered the possibility of adapting Kratos to deal with non-standard characters everywhere (for example inside the mdpa files), it is, all the strings in the code. But this option was rejected in the end, agreeing that file names might be enough.

**Comment by maceligueta** [comment]
Linux does not like this implementation, tomorrow I will insist a bit more.

**Comment by maceligueta** [comment]
Closing until it works in linux in my pc.

**Comment by loumalouomega** [comment]
@maceligueta can we delete the branch?

**Comment by maceligueta** [comment]
Yes, I completely forgot about this.

**Pull request closed without merging**